common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -65,7 +52,7 @@ def connect_to_mongodb(
             kwargs['read_preference'] = read_preference
 
     mongo_conn = pymongo.database.Database(
-        mongo_client_class(
+        pymongo.MongoClient(
```

Comment:
Looks great! From the pymongo docs here:
https://pymongo.readthedocs.io/en/3.10.1/api/pymongo/mongo_client.html
```
Changed in version 3.0: MongoClient is now the one and only client class for a standalone server, 
mongos, or replica set. It includes the functionality that had been split into MongoReplicaSetClient:
it can connect to a replica set, discover all its members, and monitor the set for stepdowns, 
elections, and reconfigs.
```
edx-platform now uses v3.10.1 of pymongo:
https://github.com/edx/edx-platform/blob/08e028ca6e8605bd30d8647a33571f97582efce7/requirements/edx/base.txt#L745

I also checked the rest of the code base - and no other location uses `MongoReplicaSetClient`.